### PR TITLE
feat: mixpanel deletion api

### PR DESCRIPTION
**Pull request by Gauravudia**
## Description of the change

> Add support of Create Task Deletion API

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/v0/destinations/mp/deleteUsers.js [hunk 1]
```diff
@@ -68,6 +64,81 @@ const userDeletionHandler = async (userAttributes, config) => {
...
+    Object.keys(chunkedUserIds).map(async (complianceType) => {
+      const batchEvents = chunkedUserIds[complianceType];
+      await Promise.all(
+        batchEvents.map(async (batchEvent) => {
+          const request = {
+            distinct_ids: batchEvent,
+            compliance_type: complianceType,
+          };
+          const deletionResponse = await httpPOST(endpoint, request, { headers });
+          const handledDelResponse = processAxiosResponse(deletionResponse);
```
Comment (sanpj2292) [inline src/v0/destinations/mp/deleteUsers.js:102]: Can we use `handleHttpRequest` method instead of `httpPOST` ?

**Review by ItsSudip — DISMISSED** [review]

File: src/v0/destinations/mp/deleteUsers.js (hunk 1 above)
Comment (Gauravudia) [reply src/v0/destinations/mp/deleteUsers.js]: updated

**Review by ItsSudip — APPROVED** [review]

File: src/v0/destinations/mp/deleteUsers.js [hunk 2]
```diff
@@ -62,12 +110,31 @@ const userDeletionHandler = async (userAttributes, config) => {
...
+  }
+
+  if (config?.userDeletionApi === 'task') {
+    return createDeletionTask(userAttributes, config);
+  }
+
+  return deleteProfile(userAttributes, config);
+};
+
 const processDeleteUsers = (event) => {
```
Comment (sanpj2292) [inline src/v0/destinations/mp/deleteUsers.js:138]: Should there be an `async` to this function ?
And also we might need to `await` on `userDeletionHandler` right ?

File: src/v0/destinations/mp/deleteUsers.js
Comment (sanpj2292) [inline src/v0/destinations/mp/deleteUsers.js:1]: [Optional] This is more optimal and easily readable IMHO
```suggestion
// deleteProfile method
const data = userAttributes.filter(attr => attr.userId).map((userAttribute) => {
    // Dropping the user if userId is not present
      return {
        $distinct_id: userAttribute.userId,
        ...defaultValues,
      };
  });
```

**Review by yashasvibajpai — APPROVED** [review]

File: src/v0/destinations/mp/deleteUsers.js (hunk 2 above)
Comment (Gauravudia) [reply src/v0/destinations/mp/deleteUsers.js]: Yes, async/ await is needed here. It is missing in some of the deleteUser implementations. I will address those in a separate PR. In this PR i have added the async/await for mixpanel

File: src/v0/destinations/mp/deleteUsers.js
Comment (Gauravudia) [reply src/v0/destinations/mp/deleteUsers.js:1]: Incorporated the changes

File: src/v0/destinations/mp/deleteUsers.js (hunk 2 above)
Comment (Gauravudia) [reply src/v0/destinations/mp/deleteUsers.js]: Added for other destinations in [this](https://github.com/rudderlabs/rudder-transformer/pull/2183) PR.

Comment (sanpj2292) [reply src/v0/destinations/mp/deleteUsers.js]: 🆒

**Review by sanpj2292 — APPROVED** [review]

File: test/deleteUsers/data/mp/handler_input.json [hunk 3]
```diff
@@ -3083,5 +3083,53 @@
...
     }
+  },
+  {
+    "request": {
+      "body": [
+        {
+          "destType": "MP",
+          "userAttributes": [
+            {
+              "userId": "rudder1"
```
Comment (ItsSudip) [inline test/deleteUsers/data/mp/handler_input.json:3094]: Have we considered the case where we will get other user attributes inside the object?

Comment (Gauravudia) [reply test/deleteUsers/data/mp/handler_input.json:3094]: we are not using other attributes.

**Review by utsabc — APPROVED** [review]

File: test/deleteUsers/data/mp/handler_input.json (hunk 3 above)
Comment (ItsSudip) [reply test/deleteUsers/data/mp/handler_input.json:3094]: What if another destination's deletion function is using other attributes and the same event came in Mixpanel?

Comment (Gauravudia) [reply test/deleteUsers/data/mp/handler_input.json:3094]: Then we will only take `userId` for calculation.

**Review by ItsSudip — APPROVED** [review]

**Pull request merged**